### PR TITLE
Fix MSVC build break

### PR DESCRIPTION
**Pull request by sanmut**

### CODE REVIEW

File: source/core/Math.cpp [hunk 1]
```diff
@@ -12,6 +12,10 @@ SDG
  */
 
 #include <cstdlib>
+
+#ifdef _MSC_VER
```
Comment (spathiwa) [inline source/core/Math.cpp]: I think this belongs in BuildConfig.h and not here.

Comment (sanmut) [inline source/core/Math.cpp]: How about the following in BuildConfig.h (and use kVireoCompiler_MSVC here)?
#if defined(_MSC_VER)
	#define kVireoCompiler_MSVC 1
#endif
I think we want to distinguish OS vs Compiler.

File: source/include/BuildConfig.h [hunk 2]
```diff
@@ -28,6 +28,10 @@ SDG
 	#define kVireoOS_windows 1
 #endif
 
+#ifdef _MSC_VER
```
Comment (cglzaguilar) [inline source/include/BuildConfig.h]: Close to the bottom of the file, there is already a Vireo windows section, shouldn't be put it there?

 #elif (kVireoOS_windows)
 #if (_MSC_VER < 1900)		 #if (_MSC_VER < 1900)
     #define snprintf _snprintf		     #define snprintf _snprintf
 #endif		 #endif

Comment (sanmut) [inline source/include/BuildConfig.h]: Will do.

**Pull request closed without merging**